conans/test/unittests/client/toolchain/autotools/autotools_deps_test.py

```diff
@@ -85,5 +94,54 @@ def test_foo():
                                  '-Wl,-rpath,"/path/to/folder_dep2/one/lib/path/dep2" ' \
                                  '--sysroot=/path/to/folder/dep1'
 
+        assert deps.environment["CXXFLAGS"] == 'dep1_a_cxx_flag dep2_a_cxx_flag ' \
+                                               '--sysroot=/path/to/folder/dep1'
+        assert deps.environment["CFLAGS"] == 'dep1_a_c_flag dep2_a_c_flag ' \
+                                             '--sysroot=/path/to/folder/dep1'
+
+
+def test_adjust_before_generate(dep1, dep2):
+
+    with mock.patch('conans.ConanFile.dependencies', new_callable=mock.PropertyMock) as mock_deps:
+        req1 = Requirement(ConanFileReference.loads("dep1/1.0"))
+        req2 = Requirement(ConanFileReference.loads("dep2/1.0"))
+        deps = OrderedDict()
+        deps[req1] = ConanFileInterface(dep1)
+        deps[req2] = ConanFileInterface(dep2)
+        mock_deps.return_value = ConanFileDependencies(deps)
+        consumer = ConanFile(Mock(), None)
+        consumer.settings = MockSettings(
+            {"build_type": "Release",
+             "arch": "x86",
+             "compiler": "gcc",
+             "compiler.libcxx": "libstdc++11",
+             "compiler.version": "7.1",
+             "cppstd": "17"})
+        deps = AutotoolsDeps(consumer)
+
+        # We can access the cpp_info and modify it
+        deps.cpp_info.includedirs = ["/other/includedir"]
+
+        # Later we can access to the flags and modify them
+        deps.cpp_flags.append("NewCppFlag")
```

Comment:
I am not sure that adding flags is a use case for ``AutotoolsDeps``, it should to go ``AutotoolsToolchain``.
I think that some control over the dependencies flags, like enabling/disabling/filter some or all of the dependencies flags might be useful or necessary, but adding a new flag? Probably not here?

Reply:
Well, that is a test, the cpp_flags are the ones filled from the deps information.